packages/nestjs-role/src/services/role.service.ts

```diff
@@ -308,4 +312,15 @@ export class RoleService implements RoleServiceInterface {
       throw new RoleAssignmentNotFoundException(assignment);
     }
   }
+  /**
+   * Create a repository proxy for the given assignment repository
+   *
+   * @internal
+   * @param assignmentRepo - The role assignment repository
+   */
+  private createRepoProxy(
+    assignmentRepo: Repository<RoleAssignmentEntityInterface>,
+  ): RepositoryProxy<RoleAssignmentEntityInterface> {
+    return new RepositoryProxy<RoleAssignmentEntityInterface>(assignmentRepo);
+  }
```

Comment:
This still creates a new class on every call. It needs to be instantiated one time when the class is created.

Reply:
nm, i was thinking of EntityMangagerProxy